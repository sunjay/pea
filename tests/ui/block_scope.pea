fn main() {
    let b = 44;
    if true {
        let a = 3;
        let z = a * 2;
    } else {
        let a = -3;
        // values from the other branch aren't available here
        println!(z);
    }

    // values from within the blocks of an if aren't available here
    println!(b + a);
}
