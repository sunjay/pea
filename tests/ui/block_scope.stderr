[tests/ui/block_scope.pea:9:18] error: cannot find name `z` in this scope

[tests/ui/block_scope.pea:13:18] error: cannot find name `a` in this scope

error: aborting due to 2 previous errors

